docs/common_use_cases.md

```diff
@@ -0,0 +1,14 @@
+# Common Use Cases
+
```

Comment:
Shouldn't the first "common use case" be `dbt`?
- Framing things in terms of `dbt`, `joindiff`, and `hashdiff` would be consistent with what you put [here](https://github.com/datafold/data-diff/pull/471/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R118). 
- The `dbt` use case should be the _first_ thing people read about, since we expect users to increasingly be coming from the dbt community.
- We don't want dbt users to mistakenly use `joindiff` when inspecting differences between branches, making sure code results in only expected changes, etc. (which are the "common use cases" for `joindiff`).

Reply:
Sure, want me to add a commit to this branch?

Reply:
Or, it could be a follow-on PR, so as not to slow this down.

Reply:
That’s fine too. I don’t think we are currently linking to the use cases file anyway

Reply:
OK, then I'll do it as a follow-on.